Tasks/NodeToolV0/nodetool.ts

```diff
@@ -47,36 +48,50 @@ interface INodeVersion {
 //      PATH = cacheDir + PATH
 //
 async function getNode(versionSpec: string, checkLatest: boolean) {
+    let installedArch = osArch;
     if (toolLib.isExplicitVersion(versionSpec)) {
         checkLatest = false; // check latest doesn't make sense when explicit version
     }
 
     // check cache
     let toolPath: string;
     if (!checkLatest) {
-        toolPath = toolLib.findLocalTool('node', versionSpec, osArch);
+        toolPath = toolLib.findLocalTool('node', versionSpec, installedArch);
+
+        // In case if it's darwin arm and toolPath is empty trying to find x64 version
+        if (!toolPath && isDarwinArm(osPlat, installedArch)) {
+            toolPath = toolLib.findLocalTool('node', versionSpec, 'x64');
+        }
     }
 
     if (!toolPath) {
         let version: string;
         if (toolLib.isExplicitVersion(versionSpec)) {
             // version to download
             version = versionSpec;
-        }
-        else {
+        } else {
             // query nodejs.org for a matching version
-            version = await queryLatestMatch(versionSpec);
+            version = await queryLatestMatch(versionSpec, installedArch);
+
+            if (!version && isDarwinArm(osPlat, installedArch)) {
+                // nodejs.org does not have an arm64 build for macOS, so we fall back to x64
```

Comment:
As of Node 16, there are arm64 builds for macOS, [see](https://nodejs.org/dist/v16.0.0/)

Reply:
yup, and it will be found in 
`version = await queryLatestMatch(versionSpec, installedArch);` on line 74, if the node version for arm64 is not found (for example for node10) the script will check that rosetta is installed and run and try to find the version for x64 arch to run it under rosetta